bootstrap/remote/verify.py

```
@@ -305,6 +311,158 @@ def verify_blob_integrity(self) -> list[Issue]:
+        """For each channel head, verify every `snapshot_hash` in the head's
+        `history.pb2` resolves to an existing resource-snapshot directory.
+        Missing → error (blocks verify exit)."""
+        issues: list[Issue] = []
+        registry = self.head_store.get_registry()
+
+        from bootstrap.remote.models import ServerHistory
+
+        for channel_name in registry.channels:
+            try:
+                head = self.head_store.get_head(channel_name)
+            except Exception as exc:
```

Ruff (0.15.18)

[warning] Do not catch blind exception: `Exception`

(BLE001)